src/global/BaseRandomTest.java

```diff
@@ -0,0 +1,105 @@
+package global;
+
+import global.variables.RandomDouble;
+import global.variables.RandomInteger;
+import global.variables.RandomVariable;
+import global.variables.Regexable;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.RepetitionInfo;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+
+import static global.utils.RandomUtil.followsUniformDistribution;
+import static global.utils.RegexUtil.*;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public abstract class BaseRandomTest extends BaseTest {
+    static Map<Integer, ArrayList<?>> randomMap = new HashMap<>();
+    static boolean alreadyFailed = false;
+
+    public void resetRandomMap() {
+        randomMap = null;
+        randomMap = new HashMap<>();
+    }
+
+    public void initializeRandomNumberMaps() {
+        resetRandomMap();
+        // TODO: need to adapt to multi line output
+        Regexable[] regexSentence = getRegexSentence();
+        for (int i = 0; i < regexSentence.length; i++) {
+            Regexable regexClause = regexSentence[i];
+            if (!(regexClause instanceof RandomVariable)) continue;
+            RandomVariable<?> randomVariable = (RandomVariable<?>) regexClause;
+
+            int regexGroup = i + 1;
+            randomMap.put(regexGroup, randomVariable.createArrayList());
+        }
+    }
+
+    public void addValueToMapList(RandomVariable<?> randomVariable, int mapKey, Object newListValue) {
+        randomVariable.addValueToMapListHelper((HashMap<Integer, ?>) randomMap, mapKey, newListValue);
+    }
+
+    // TODO: choose number
+    @RepeatedTest(500)
+    public void testRandomNumbers(RepetitionInfo repetitionInfo) {
+        if (repetitionInfo.getCurrentRepetition() == 1) {
+            initializeRandomNumberMaps();
+        }
+
+        if (alreadyFailed) fail();  // TODO: No message (?)
+
+        String output = getOutput();
+        // TODO: not always println though (in general how do we handle multi line outputs?) - arrays for each line in regex sentence?
+        Matcher matcher = getMatches(output, processRegexForPrintlnOutput(combineRegex(getRegexSentence())));
+        if (matcher.find()) {
+            for (int i = 0; i < getRegexSentence().length; i++) {
+                Regexable regexClause = getRegexSentence()[i];
+                if (!(regexClause instanceof RandomVariable)) continue;
+                RandomVariable<?> randomVariable = (RandomVariable<?>) regexClause;
```

Comment:
I've seen the exact same conditional being used multiple times, so would it be better to make a helper function for this?

Reply:
I feel like it'd be a waste to write a function that just does a 1 line operation yknow? Unless we think it might be potentially more complex in the future